neqo-transport/src/tracking.rs

```diff
@@ -799,34 +854,135 @@ mod tests {
             assert!(rp.ack_time().is_none());
             assert!(!rp.ack_now(*NOW));
 
-            // Any packet will be acknowledged straight away.
+            // Any packet in these spaces is acknowledged straight away.
             rp.set_received(*NOW, 0, true);
             assert_eq!(Some(*NOW), rp.ack_time());
             assert!(rp.ack_now(*NOW));
         }
     }
 
     #[test]
-    fn ooo_no_ack_delay() {
-        for space in &[
-            PacketNumberSpace::Initial,
-            PacketNumberSpace::Handshake,
-            PacketNumberSpace::ApplicationData,
-        ] {
-            let mut rp = RecvdPackets::new(*space);
-            assert!(rp.ack_time().is_none());
-            assert!(!rp.ack_now(*NOW));
+    fn ooo_no_ack_delay_new() {
+        let mut rp = RecvdPackets::new(PacketNumberSpace::ApplicationData);
+        assert!(rp.ack_time().is_none());
+        assert!(!rp.ack_now(*NOW));
 
-            // Any OoO packet will be acknowledged straight away.
-            rp.set_received(*NOW, 3, true);
-            assert_eq!(Some(*NOW), rp.ack_time());
-            assert!(rp.ack_now(*NOW));
-        }
+        // Anything other than packet 0 is acknowledged immediately.
+        rp.set_received(*NOW, 1, true);
+        assert_eq!(Some(*NOW), rp.ack_time());
+        assert!(rp.ack_now(*NOW));
+    }
+
+    #[test]
+    fn ooo_no_ack_delay_gap() {
+        let mut rp = RecvdPackets::new(PacketNumberSpace::ApplicationData);
+        assert!(rp.ack_time().is_none());
+        assert!(!rp.ack_now(*NOW));
+
+        // Packet number 0 causes delayed acknowledgment.
+        rp.set_received(*NOW, 0, true);
+        assert_ne!(Some(*NOW), rp.ack_time());
+
+        // A gap causes immediate acknowledgment.
+        rp.set_received(*NOW, 2, true);
```

Comment:
You are right, this test is just defunct now.  Gaps don't cause immediate acknowledgment.  Is there a test we should be using instead?  I can't think of one, which means that this can just be deleted.